surfacers/datadog/datadog.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datadog
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"time"
+
+	datadog "github.com/DataDog/datadog-api-client-go/api/v1/datadog"
+	"github.com/google/cloudprober/logger"
+	"github.com/google/cloudprober/metrics"
+	configpb "github.com/google/cloudprober/surfacers/datadog/proto"
+	"google.golang.org/protobuf/proto"
+)
+
+/*
+	The datadog surfacer presents EventMetrics to the datadog SubmitMetrics APIs,
+	using the config passed in.
+
+	Some EventMetrics are not supported here, as the datadog SubmitMetrics API only
+	supports float64 type values as the metric value.
+*/
+
+// Datadog API limit for metrics included in a SubmitMetrics call
+const datadogMaxSeries int = 20
+
+// All metrics are of type gauge
+const gauge string = "gauge"
+
+// DDSurfacer implements a datadog surfacer for datadog metrics.
+type DDSurfacer struct {
+	c                 *configpb.SurfacerConf
+	writeChan         chan *metrics.EventMetrics
+	client            *datadog.APIClient
+	l                 *logger.Logger
+	ignoreLabelsRegex *regexp.Regexp
+	prefix            string
+	// A cache of []*datadog.Series, used for batch writing to datadog
+	ddSeriesCache []datadog.Series
+}
+
+func (dd *DDSurfacer) receiveMetricsFromEvent(ctx context.Context) {
+	for {
+		select {
+		case <-ctx.Done():
+			dd.l.Infof("Context canceled, stopping the surfacer write loop")
+			return
+		case em := <-dd.writeChan:
+			dd.emToDDSeries(ctx, em)
+		}
+	}
+}
+
+func (dd *DDSurfacer) emToDDSeries(ctx context.Context, em *metrics.EventMetrics) {
```

Comment:
Since this function doesn't return anything, can you please rename it to 'recordEventMetrics'. That will also be consistent with cloudwatch.